warehouse/transformer/datatype.go

```diff
@@ -93,3 +96,45 @@ func convertValIfDateTime(val any, colType string) any {
 	}
 	return val
 }
+
+func convertToFloat64IfInteger(val any) any {
+	switch v := val.(type) {
+	case int:
+		return float64(v)
+	case int8:
+		return float64(v)
+	case int16:
+		return float64(v)
+	case int32:
+		return float64(v)
+	case int64:
+		return float64(v)
+	case uint:
+		return float64(v)
+	case uint8:
+		return float64(v)
+	case uint16:
+		return float64(v)
+	case uint32:
+		return float64(v)
+	case uint64:
+		return float64(v)
+	}
+	return val
+}
+
+func convertToSliceIfViolationErrors(val any) any {
+	if validationErrors, ok := val.([]types.ValidationError); ok {
+		result := make([]any, len(validationErrors))
+		for i, e := range validationErrors {
+			result[i] = map[string]any{
+				"type":     e.Type,
+				"message":  e.Message,
+				"meta":     lo.MapValues(e.Meta, func(value, _ string) any { return value }),
```

Comment:
Why not assign `e.Meta` directly instead of using `MapValues` ? Is it to create a new map ?

Reply:
Meta is map[string]string. We need map[string]any.